packages/@aws-cdk/aws-lambda/lib/event-source-mapping.ts

```diff
@@ -174,6 +187,10 @@ export class EventSourceMapping extends cdk.Resource implements IEventSourceMapp
       }
     });
 
+    if (props.tumblingWindow && props.tumblingWindow.toSeconds() > 900) {
```

Comment:
you need to check that tumbling window is not a token

```suggestion
    if (props.tumblingWindow && !props.tumblingWindow.isUnresolved() && props.tumblingWindow.toSeconds() > 900) {
```

Reply:
please add a corresponding unit test.